platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/internal/ConsensusRound.java

```diff
@@ -109,69 +90,67 @@ public int getNumAppTransactions() {
      *
      * @return the list of events in this round
      */
-    public List<EventImpl> getConsensusEvents() {
+    public @NonNull List<EventImpl> getConsensusEvents() {
         return consensusEvents;
     }
 
     /**
      * @return the consensus generations when this round reached consensus
      */
-    public GraphGenerations getGenerations() {
+    public @NonNull GraphGenerations getGenerations() {
         return generations;
     }
 
+    /**
+     * @return a snapshot of consensus at this consensus round
+     */
+    public @NonNull ConsensusSnapshot getSnapshot() {
+        return snapshot;
+    }
+
     /**
      * @return the number of events in this round
      */
     public int getNumEvents() {
         return consensusEvents.size();
     }
 
-    /**
-     * {@inheritDoc}
-     */
+    /** {@inheritDoc} */
     @Override
-    public Iterator<ConsensusEvent> iterator() {
+    public @NonNull Iterator<ConsensusEvent> iterator() {
         return new TypedIterator<>(consensusEvents.iterator());
     }
 
-    /**
-     * {@inheritDoc}
-     */
+    /** {@inheritDoc} */
     @Override
     public long getRoundNum() {
-        return roundNum;
+        return snapshot.round();
     }
 
-    /**
-     * {@inheritDoc}
-     */
+    /** {@inheritDoc} */
     @Override
     public boolean isEmpty() {
         return consensusEvents.isEmpty();
     }
 
-    /**
-     * {@inheritDoc}
-     */
+    /** {@inheritDoc} */
     @Override
     public int getEventCount() {
         return consensusEvents.size();
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @NonNull
+    /** {@inheritDoc} */
     @Override
-    public Instant getConsensusTimestamp() {
-        return consensusTimestamp;
+    public @NonNull Instant getConsensusTimestamp() {
+        return snapshot.consensusTimestamp();
     }
 
     /**
      * @return the last event of this round, or null if this round is not complete
+     * @deprecated a round might not have any events, the code should not expect this to be non-null
```

Comment:
Is there a ticket in the backlog to eliminate this method?

Reply:
added #8605